### PR TITLE
Add link to Gmail exporter

### PR DESCRIPTION
**Pull request by jamesread**
Signed-off-by: James Read contact@jread.com

### CODE REVIEW

**Comment by brian-brazil** [comment]
I think this better belongs under API.

The label name should be a label rather than part of the metric name. `_total` also indicates a counter, so you should remove it from the gauge.

File: content/docs/instrumenting/exporters.md [hunk 1]
```diff
@@ -179,6 +179,7 @@ wide variety of JVM-based applications, for example [Kafka](http://kafka.apache.
    * [Dnsmasq exporter](https://github.com/google/dnsmasq_exporter)
    * [eBPF exporter](https://github.com/cloudflare/ebpf_exporter)
    * [Ethereum Client exporter](https://github.com/31z4/ethereum-prometheus-exporter)
+   * [Gmail exporter](https://github.com/jamesread/prometheus-gmail-exporter/)
```
Comment (roidelapluie) [inline content/docs/instrumenting/exporters.md]:
```suggestion
   * [Gmail exporter](https://github.com/jamesread/prometheus-gmail-exporter)
```

**Comment by jamesread** [comment]
Hey @brian-brazil , thanks for the feedback.

> I think this better belongs under API.

Okay doke, shall I just move the line in a commit on top of this one? 

> The label name should be a label rather than part of the metric name

Thanks, it is useful to have a steer on what's more idiomatic for Prometheus. Does this format look good to you? (haven't committed these changes yet, want to check they look good first).

```
# HELP gmail_total desc
# TYPE gmail_total gauge
gmail_total{gmail_label_id="INBOX",gmail_label_name="INBOX"} 44358.0
gmail_total{gmail_label_id="Label_33",gmail_label_name=">/0. Triage"} 159.0
# HELP gmail_unread desc
# TYPE gmail_unread gauge
gmail_unread{gmail_label_id="INBOX",gmail_label_name="INBOX"} 43.0
gmail_unread{gmail_label_id="Label_33",gmail_label_name=">/0. Triage"} 0.0
```

I decided on `gmail_label_name` rather than just `label_name`, gmail labels could get confused with Prometheus labels by users. Maybe I'm over-thinking that though. 

> _total also indicates a counter, so you should remove it from the gauge.

There is now `gmail_total` and `gmail_unread`. Are you suggesting the metric name should simply be `gmail`, and have a `count_type="total"` label or something? 

@roidelapluie , sure, I'll remove the trailing / :-)

**Comment by brian-brazil** [comment]
> Okay doke, shall I just move the line in a commit on top of this one?

Whatever is easiest for you, just watch for the DCO.

> Are you suggesting the metric name should simply be gmail, and have a count_type="total" label or something?

gmail_emails and gmail_emails_unread would be my first thought.

If the label names are unique (can you have dupes?) then you could drop the id.

**Comment by RichiH** [comment]
You write on medium that you update every five minutes. Anything above 120s
can lead to degraded experience for your users.

Sent by mobile; please excuse my brevity.

**Comment by brian-brazil** [comment]
Based on the comments in the code there is rate limiting on the API which causes issues, so the background updates are likely a good tradeoff in this particular case. I'd also hope noone gets quite that much email that it'd matter :)

This can be merged once it's in the right section.

**Comment by jamesread** [comment]
Hey @RichiH . 

> You write on medium that you update every five minutes. Anything above 120s
can lead to degraded experience for your users. 

Yep. I've sort-of designed in 3/4 precautions with this in mind already. 

1) Emit a warning if people default to all labels in their account (assuming most people will just "watch" 1 or 2 labels). 
https://github.com/jamesread/prometheus-gmail-exporter/blob/22a16894a375e19a36f0cdd8ec72f0ef870678b6/gmail-exporter.py#L86

2) Individual HTTP requests are logged by default, so people can see the number of requests / time. 

3) The default update delay is 5 minutes (300 seconds). See https://github.com/jamesread/prometheus-gmail-exporter/blob/22a16894a375e19a36f0cdd8ec72f0ef870678b6/gmail-exporter.py#L156 . This of course can be turned up / down as needed. 

4) Catch on API "hickups". I was getting 404's and stuff from Gmail in one in every couple of hundred of requests. If Gmail starts sending 400's or something for rate limiting, the script should fail gracefully. https://github.com/jamesread/prometheus-gmail-exporter/blob/22a16894a375e19a36f0cdd8ec72f0ef870678b6/gmail-exporter.py#L121 

@brian-brazil ; 

> This can be merged once it's in the right section.

Thanks! Will commit the changes as suggested here back to my repo, and move this doc line in a commit - will do both when the day job gives me some time - ASAP :-)

**Comment by jamesread** [comment]
Hey @brian-brazil , sorry for the delay. 

Fingers crossed that with 6793348 this PR is OK now :)

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**